src/Npgsql/ConnectorPool.cs

```diff
@@ -129,7 +129,7 @@ internal ConnectorPool(NpgsqlConnectionStringBuilder settings, string connString
                 throw new ArgumentException($"Connection can't have ConnectionIdleLifetime {connectionIdleLifetime} under ConnectionPruningInterval {_pruningSamplingInterval}");
 
             _pruningTimer = new Timer(PruningTimerCallback, this, Timeout.Infinite, Timeout.Infinite);
-            _pruningSampleSize = DivideRoundingUp(connectionIdleLifetime.Seconds, pruningSamplingInterval.Seconds);
+            _pruningSampleSize = DivideRoundingUp(settings.ConnectionIdleLifetime, settings.ConnectionPruningInterval);
```

Comment:
Nit: seems like the local variables (connectionIdleLifetime, pruningSamplingInterval) are now only used for error-checking above... For slightly cleaner code maybe we can just copy settings.ConnectionIdleLifetime to connectionIdleLifetime (without TimeSpan.FromSeconds) and use that everywhere?

Otherwise LGTM

Reply:
Not entirely, pruningSamplingInterval is stored as TimeSpan in _pruningSamplingInterval too.
It did go through my mind to remove the uses but I decided against it when I saw that.

Reply:
OK, no problem.